docs/getting_started/index.md

```diff
@@ -133,21 +127,95 @@ Name: <|{input_name}|input|>
 Message: <|{message}|text|>
 """
 
+input_name = "Taipy"
+message = None
+
+
+def submit_scenario(state):
+    state.scenario.input_name.write(state.input_name)
+    state.scenario.submit(wait=True)
+    state.message = scenario.message.read()
+
+
 if __name__ == "__main__":
     Core().run()
-    hello_scenario = tp.create_scenario(scenario_cfg)
+    scenario = tp.create_scenario(scenario_cfg)
     Gui(page).run()
 ```
 
 To get the complete example, including the user interface code, download it here:
 <a href="./hello_world.py" download>`hello_world.py`</a>
 
-TO DO: explain the code above and result below
+Now, let’s explain the key elements of this code:
+
+```python
+from taipy import Gui
+import taipy as tp
+
+page = """
+Name: <|{input_name}|input|>
+<|submit|button|on_action=submit_scenario|>
+Message: <|{message}|text|>
+"""
+```
+
+**Pages**
+
+Taipy pages can be defined in multiple ways: Markdown, Html or Python. The `page` 
+variable is a Markdown representation of the user interface. 
+It uses standard Markdown syntax as well as visual elements.
+
+**Visual elements**
+
+Taipy offers various visual elements that can interact with your Python variables and 
+environment. It allows you to display variables, modify them, and interact with the application.
+
+Here is a generic example of one: `<|{variable}|visual_element_type|...|>`. *variable* is 
+the main property of the visual element and is usually what is displayed or modified through the visual element.
+
+In our initial example: 
+
+- *input_name* is bound to an input and text field, allowing the user’s input to be directly stored in the *input_name* variable.
+
+- *message* is connected to a text field, allowing you to display changing content to the user.
+
+#### Interactivity Through Actions
+
+Actions, like `on_action=submit_scenario`, allow visual elements like buttons to trigger specific functions, enhancing the interactivity of the application.
+
+```python
+def submit_scenario(state):
+    scenario.input_name.write(state.input_name)
+    scenario.submit()
+    state.message = scenario.message.read()
+```
+
+Every callback, including `submit_scenario`, utilizes state as the first parameter. 
+This state represents a user's connection and is used to read and set variables while 
+the user is interacting with the program. It makes it simple for Taipy to handle multiple users simultaneously.
+
+While *scenario* is available to everyone using the program, *state.input_name* and *state.input* are unique to the user 
```

Comment:
"unique to the user who defines them" -> "specific to the user who interacts with them"